corehq/apps/reports/standard/users/reports.py

```diff
@@ -116,21 +149,56 @@ def rows(self):
             self.pagination.start:self.pagination.start + self.pagination.count
         ]
         for record in records:
-            yield _user_history_row(record, self.timezone)
+            yield _user_history_row(record, self.domain, self.timezone)
 
 
-def _user_history_row(record, timezone):
+def _user_history_row(record, domain, timezone):
     return [
         cached_user_id_to_username(record.user_id),
         cached_user_id_to_username(record.changed_by),
         _get_action_display(record.action),
         record.details['changed_via'],
         record.message,
-        json.dumps(record.details['changes']),
+        _details_cell(record.details['changes'], domain),
         ServerTime(record.changed_at).user_time(timezone).ui_string(USER_DATETIME_FORMAT),
     ]
 
 
+def _details_cell(changes, domain):
+    def _html_list(changes, unstyled=True):
+        items = []
+        for key, value in changes.items():
+            if isinstance(value, dict):
+                value = _html_list(value, unstyled=unstyled)
+            elif isinstance(value, list):
+                value = format_html(", ".join(value))
+            else:
+                value = format_html(str(value))
+            items.append("<li>{}: {}</li>".format(key, value))
```

Comment:
should this also use `format_html`?

Reply:
It can't because `value` might be HTML, but it should be all right because `value` goes through `format_html` in the lines above.